var/spack/repos/builtin/packages/lemon/package.py

```diff
@@ -17,36 +17,20 @@ class Lemon(CMakePackage):
 
     version("1.3.1", sha256="71b7c725f4c0b4a8ccb92eb87b208701586cf7a96156ebd821ca3ed855bad3c8")
 
+    # coin variant currently fails due to a missing dependency library
     variant("coin", default=False, description="Enable Coin solver backend")
     variant("ilog", default=False, description="Enable ILOG (CPLEX) solver backend")
     variant("glpk", default=True, description="Enable GLPK solver backend")
     # soplex not mentioned in docs but shown in cmakecache
     variant("soplex", default=False, description="Enable SOPLEX solver backend")
 
-    variant(
-        "lp_backend",
-        default="glpk",
-        description="Set default LP solver backend",
-        values=("glpk", "cplex", "clp", "soplex"),
-        multi=False,
-    )
-    variant(
-        "mip_backend",
-        default="glpk",
-        description="Set default MIP solver backend",
-        values=("glpk", "cplex", "cbc"),
-        multi=False,
-    )
 
     depends_on("glpk", when="+glpk")
-    # LP solver
-    depends_on("glpk", when="lp_backend=glpk")
-    depends_on("cplex", when="lp_backend=cplex")
-    depends_on("clp", when="lp_backend=clp")
-    # MIP solver
-    depends_on("glpk", when="mip_backend=glpk")
-    depends_on("cplex", when="mip_backend=cplex")
-    depends_on("cbc", when="mip_backend=cbc")
+    depends_on("cplex", when="+ilog")
+    depends_on("coinutils", when="+coin") # just a guess
```

Comment:
Do you know how to confirm whether this is the right package?

Reply:
It's seems they're strictly related.

Reply:
it's a direct dependency of `cbc` package, also referred in this packagefile.

Reply:
`+coin` variant is currently not building due to error:
```
  >> 63    CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
     64    Please set them or make sure they are set and tested correctly in the CMake files:
     65    COIN_BZ2_LIBRARY (ADVANCED)
     66        linked by target "lemon" in directory /cache/sbulut/spack-stage/spack-stage-lemon-1.3.1-cbkh7hrng7fqhfrcoothdx5ezkqeig4c/spack-src/lemon
     67        linked by target "lemon" in directory /cache/sbulut/spack-stage/spack-stage-lemon-1.3.1-cbkh7hrng7fqhfrcoothdx5ezkqeig4c/spack-src/lemon
     68        linked by target "lp_test" in directory /cache/sbulut/spack-stage/spack-stage-lemon-1.3.1-cbkh7hrng7fqhfrcoothdx5ezkqeig4c/spack-src/test
     69        linked by target "mip_test" in directory /cache/sbulut/spack-stage/spack-stage-lemon-1.3.1-cbkh7hrng7fqhfrcoothdx5ezkqeig4c/spack-src/test
```

Reply:
I am only interested in `+glpk` variant really. We can probably leave this as TODO.